### ISSUE TITLE
`native_batch_norm` silently produces incorrect 3rd+ order derivatives in training mode with `create_graph=True`

### ISSUE THREAD

**Issue by f-dangel** [pytorch/pytorch#186256]
### 🐛 Describe the bug

Repeated differentiation (`create_graph=True`) through the fused `native_batch_norm` in **training mode** silently produces **incorrect derivatives from 3rd order onward**. Orders 0–2 are correct; order ≥3 diverges from the mathematically correct value by large amounts (not floating-point noise). Because no error is raised, this silently corrupts any computation relying on 3rd+ order derivatives of batch norm.

This is the same root cause as #175754 for `native_layer_norm`: the saved `mean`/`invstd` tensors handed to the fused double-backward have no autograd history connecting them to the input, so when autograd differentiates *through* the double backward for 3rd+ order derivatives it treats them as constants.

It only affects **training mode** (where `mean`/`invstd` depend on the input); eval mode is an affine map with frozen statistics and is correct to all orders.

### To reproduce

Compare the fused op against a mathematically identical decomposition from primitives, taking directional derivatives `dᵏ/dtᵏ ⟨f(x + t·v), w⟩` at `t = 0` via repeated `autograd.grad(..., create_graph=True)`:

```python
import torch
from torch.nn.functional import batch_norm

torch.manual_seed(0)
torch.set_default_dtype(torch.float64)

N, C = 8, 3
x = torch.randn(N, C)
v = torch.randn(N, C)             # direction
w = torch.randn(N, C)            # cotangent -> scalar output

running_mean, running_var = torch.zeros(C), torch.ones(C)

def fused(z):                    # fused native_batch_norm (training)
    return batch_norm(z, running_mean, running_var, None, None,
                      training=True, momentum=0.0, eps=1e-5)

def manual(z):                   # mathematically identical, from primitives
    mean = z.mean(0, keepdim=True)
    c = z - mean
    var = (c * c).mean(0, keepdim=True)
    return c * (var + 1e-5).rsqrt()

def t_derivatives(f, order):     # d^k/dt^k <f(x + t v), w> at t = 0
    t = torch.zeros((), requires_grad=True)
    g = (f(x + t * v) * w).sum()
    coeffs = []
    for k in range(order + 1):
        coeffs.append(g.detach().clone())
        if k < order:
            (g,) = torch.autograd.grad(g, t, create_graph=True)
    return coeffs

fused_d = t_derivatives(fused, 5)
manual_d = t_derivatives(manual, 5)
print("order | match | max_abs_diff")
for k in range(6):
    match = torch.allclose(fused_d[k], manual_d[k])
    print(f"  {k}   |  {str(match):5} | {(fused_d[k] - manual_d[k]).abs().max().item():.3e}")
```

### Output (should match everywhere, but does not):
```
order | match | max_abs_diff
  0   |  True  | 0.000e+00
  1   |  True  | 0.000e+00
  2   |  True  | 8.882e-16
  3   |  False | 1.258e+00
  4   |  False | 7.850e+01
  5   |  False | 2.695e+02
```

### Expected behaviour

Either the fused training-mode double-backward should produce correct 3rd+ order derivatives (matching the primitive decomposition), or — as a strict improvement until then — it should raise rather than return silently-wrong values (cf. the layer_norm proposal in #176234).

### Related

- #175754 — same bug for `native_layer_norm` (closed; root cause documented there)
- #176234 — proposed layer_norm guard; reverted, not on `main`

### Versions

```
Collecting environment information...
PyTorch version: 2.12.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 26.5.1 (arm64)
GCC version: Could not collect
Clang version: 21.0.0 (clang-2100.1.1.101)
CMake version: Could not collect
Libc version: N/A

Python version: 3.12.6 | packaged by Anaconda, Inc. | (main, Oct  3 2024, 02:26:31) [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-26.5.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Apple M5

Versions of relevant libraries:
[pip3] backpack-for-pytorch==1.7.1
[pip3] jet-for-pytorch==0.0.2.dev133+gf171ce318
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] torchvision==0.27.0
[conda] Could not collect
```

cc @ezyang @gchanan @kadeng @msaroufim @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93

**Comment by zou3519** [comment]
triage review says: layer_norm fix was to make it error out, which would be OK here

**Comment by he-yufeng** [comment]
I confirmed the current manual double-backward path has the same detached saved-statistics problem as the layer-norm case. I am preparing the accepted fail-fast direction using DelayedError, with a focused regression test and an explicit compiled-autograd skip so it does not repeat the CI failure that reverted #176234.
